### PR TITLE
Adds close method to VersionedHDF5File objects. Updates docs.

### LINKED ISSUE #45: Add close() method to VersionedHDF5File
As mentioned in https://github.com/Quansight/versioned-hdf5/pull/42

### PR DESCRIPTION
**Pull request by melissawm**
Please let me know if there's a better approach.

Closes #45

### CODE REVIEW

File: versioned_hdf5/api.py [hunk 1]
```diff
@@ -150,3 +156,32 @@ def stage_version(self, version_name: str, prev_version=None,
         except:
             delete_version(self.f, version_name, old_current)
             raise
+
+    def close(self):
+        """
+        Make sure the VersionedHDF5File object is no longer reachable.
+        """
+        if not self._closed:
+            self.__del__()
```
Comment (asmeurer) [inline versioned_hdf5/api.py:165]: Should it call self.f.close()?

**Comment by asmeurer** [comment]
What is the purpose of the `__del__` method here?

File: versioned_hdf5/api.py (hunk 1 above)
Comment (melissawm) [reply versioned_hdf5/api.py:165]: I was not sure if this should be done silently.

**Comment by melissawm** [comment]
> What is the purpose of the `__del__` method here?

It was about not forgetting to `close` the object after finishing but I know this might not be the best solution... Also since if the `h5py` file is closed the versioning data gets written, it shouldn't be a problem if we don't close the VersionedHDF5File (which is exactly what we were doing before). So yeah, I should maybe remove it.

File: versioned_hdf5/api.py [hunk 2]
```diff
@@ -150,3 +156,25 @@ def stage_version(self, version_name: str, prev_version=None,
...
+            self._closed = True
+
+    def __repr__(self):
+        """
+        Prints friendly status information.
+
+        These messages are intended to be similar to h5py messages.
+        """
+        if self._closed:
+            return "<Closed VersionedHDF5File>"
```
Comment (scopatz) [inline versioned_hdf5/api.py:177]: We could probably provide some of the same metadata in this string as below

Comment (melissawm) [reply versioned_hdf5/api.py:177]: If the `VersionedHDF5File` object is already closed, it doesn't have any metada associated with it anymore (so the object file and metadata is no longer reachable). Is there anything specifically you think might be good to include in this string?

Comment (scopatz) [reply versioned_hdf5/api.py:177]: oh ok.... nm then

**Comment by scopatz** [comment]
LGTM!

**Pull request merged**